test/generate_markdown.py

```diff
@@ -0,0 +1,159 @@
+    file_desc_dict = {}
+    if file_desc_list:
+        for fd in file_desc_list:
+            file_desc_dict[fd[0]] = fd[1]
+
+    for filename in test_files:
+        results = list(run(filename, execution_mode=ExecutionMode.TESTING))
+        base = os.path.basename(filename)
+        rule_code = re.search(r'(fail|pass)-([a-z]{3}[0-9]{3})-', base).group(2)
+        readme_path = os.path.join(os.path.dirname(__file__), 'files', rule_code, 'README.md')
+
+
+        description = file_desc_dict.get(base, " NA / Automatically generated markdown ")
+
+        if not description == " NA / Automatically generated markdown ":
+            pass
+
+        behave_output_patterns = {
+            'Instance_id': r"Instance_id=(\d+)",
+            'Expected': r"Expected=({.*?})",
+            'Observed': r"Observed=({.*?})"
+        }
+
+
+        def extract_value(pattern, text):
+            """ Utility function to apply regex search and return the matching group. """
+            match = re.search(pattern, text)
+            return match.group(1) if match else ''
+        
+        def get_parsed_behave_output(results):
+            formatted_results = []
+            for e, result in enumerate(results):
+                extracted_values = {key: extract_value(pattern, result[4][1]) for key, pattern in behave_output_patterns.items()}  
+                result_str = f"Result {e+1}: {extracted_values}"
+                formatted_results.append(result_str)
+
+            return str(" . ".join(formatted_results))
+        
+
+        content_testfile = {
+                        "File name" : base,
+                        "Expected result" : "pass" if base.startswith("pass-") else "fail",
+                        "Description": do_try(lambda: get_parsed_behave_output(results), "NA / Automatically generated markdown")
+                    }
+
+        if os.path.exists(readme_path) and os.path.getsize(readme_path) > 0:
+            with open(readme_path, 'r+') as file:
+                content = file.read()
+                file.seek(0)
+
+                markdown_table_pattern = re.compile(r'\|.*?\|\n\|[-:]+\|[-:]+', re.DOTALL)
+                has_markdown_table = bool(markdown_table_pattern.search(content))
+
+                html_table_pattern = re.compile(r'<table.*?>.*?</table>', re.DOTALL)
+                has_html_table = bool(html_table_pattern.search(content))
+
+                if has_markdown_table:
+                    df = handle_markdown_table(content)
+                elif has_html_table:
+                    df = pd.read_html(content)[0]
+                else:
+                    raise ValueError(f"The file '{readme_path}' does not contain a recognized HTML or Markdown table. Please ensure the file adheres to the expected format.")
+
+                if base not in df['File name'].values:
+                    df = update_html_markdown(df, content_testfile)
+
+                    file.write(df.to_html(index=False))
```

Comment:
```suggestion
                    file.write(df.to_markdown(index=False))
```

Kan we keep using markdown tables? Or otherwise CSV? HTML is rather difficult to hand-edit.

Reply:
All converted to markdown 
[readme.md in markdown](https://github.com/buildingSMART/ifc-gherkin-rules/pull/177/commits/34ddd9e8bb125d0d1299e4cb263fb8d909b300d1)